src/params/types.ts

```diff
@@ -69,8 +84,8 @@ export class TernaryExpression<
     this.ifFalse = ifFalse;
   }
 
-  value(): T {
-    return this.test.value() ? valueOf(this.ifTrue) : valueOf(this.ifFalse);
+  runtimeValue(): T {
+    return this.test.runtimeValue() ? valueOf(this.ifTrue) : valueOf(this.ifFalse);
```

Comment:
woah we are renaming `value` to `runtimeValue()`?

Reply:
Not from users' perspective. They'll still call value(), which will use the implementation in the base Expression class, which will do the discovery check and then call runtimeValue()